### PR TITLE
[MRG] remove py2 support

### PR DESCRIPTION
**Pull request by teonbrooks**
PR to remove py2 support from `mne-bids`. `mne-python` has removed compatibility for py2 and we depend on it. we no longer have `mne.externals.six` to maintain compatibility.

### CODE REVIEW

**Comment by teonbrooks** [comment]
closes https://github.com/mne-tools/mne-bids/issues/140

**Comment by jasmainak** [comment]
Can you add a note to the readme that we support only Python 3 now?

**Comment by monkeyman192** [comment]
I can't make review comments in the right places, but there are a few places where we should migrate to my py3-thonic code:

https://github.com/mne-tools/mne-bids/blob/e26de962b06f273a57d778852b4bc317ab3df34e/mne_bids/utils.py#L57

https://github.com/mne-tools/mne-bids/blob/e26de962b06f273a57d778852b4bc317ab3df34e/mne_bids/utils.py#L350
and
https://github.com/mne-tools/mne-bids/blob/e26de962b06f273a57d778852b4bc317ab3df34e/mne_bids/utils.py#L366
should raise `FileExistsError` (and some tests etc will need to be updated to reflect this correctly...)

These are just a few I can think of from a quick scan but there are probably more.
If you think these changes are more appropriate to be in a separate PR I can make one to update some older code for py2 compatibility to nice py3 code...

**Comment by sappelhoff** [comment]
> If you think these changes are more appropriate to be in a separate PR I can make one to update some older code for py2 compatibility to nice py3 code...

@monkeyman192, if  @teon is fine with it, you can also send a PR to `teonbrooks:py2-deletion` so that these changes are directly included here.

Also, we should be explicit about which python 3 version, ... 3.6?

**Comment by monkeyman192** [comment]
I normally spec my code for about 3.4 at the earliest (because I am still running 3.4 on one or two of my machines), however I don't think there is really anything in here that would really change across 3.x versions (other than maybe changing pathing things to path-like objects instead of simply strings which was 3.6 iirc)
It's the weekend now, so I'll look into it on monday :P

**Comment by jasmainak** [comment]
> Also, we should be explicit about which python 3 version, ... 3.6?

I think we should support all of Python 3 if it's not too much work. It's just nicer for the users if it doesn't come at an extra cost of code complexity. Otherwise, they will have to switch environments just to use mne-bids ...

File: README.md [hunk 1]
```diff
@@ -16,7 +16,8 @@ Installation
 ------------
 
 We recommend the [Anaconda](https://www.anaconda.com/download/) Python
-distribution. Besides `numpy`, `scipy`, `matplotlib`, and `pandas` (which are
+distribution. We require that you use a minimum Python version of 3.4.
```
Comment (jasmainak) [inline README.md]: I would say python 3 ... so we sync with MNE

File: .travis.yml [hunk 2]
```diff
@@ -3,7 +3,6 @@ node_js:
 language: python
 
 env:
-  - PYTHON_VERSION=2.7
   - PYTHON_VERSION=3.6
```
[the remark this answers is not in the document]
Comment (teonbrooks) [reply .travis.yml]: it seems that mne is using 3.6 even though it generically states py3. also travis and appveyor yelled about 3.0 not being found. i will be revert back to 3.6

[the remark this answers is not in the document]
Comment (jasmainak) [reply .travis.yml]: maybe we don't need to be worried about other python versions if mne is not worried ...

**Comment by teonbrooks** [comment]
> @teonbrooks you still need to address @monkeyman192 's comments

I did. https://github.com/mne-tools/mne-bids/pull/141#issuecomment-441240773. It should be a separate pr because making the code more py3 pythonic isn't a blocker for remove py2 support.
I don't know what is to be done since it is @monkeyman192's approach. I can wait for his pr to this branch before this is merged. I tend to like short, focused prs

**Comment by jasmainak** [comment]
you just need to replace all [occurences](https://github.com/mne-tools/mne-bids/blob/master/mne_bids/mne_bids.py#L278) of `errno.EEXIST` with `FileExists error. I think it's simple enough not to warrant a new PR, but as you like ...

**Comment by teonbrooks** [comment]
> you just need to replace all [occurences](https://github.com/mne-tools/mne-bids/blob/master/mne_bids/mne_bids.py#L278) of `errno.EEXIST` with `FileExists` error. I think it's simple enough not to warrant a new PR, but as you like ...

ok, added those. I just assumed that there were more py3ism that were to be implemented so I was just going to save them all. I updated to account for that find/replace.

**Comment by jasmainak** [comment]
Good to go on my end.

**Comment by sappelhoff** [comment]
thanks @teonbrooks! :-)

**Comment by monkeyman192** [comment]
> ok, added those. I just assumed that there were more py3ism that were to be implemented so I was just going to save them all. I updated to account for that find/replace.

Oh yeah, my comment was more of a general one about the possibility of updating any code that could be nicer using python 3 syntax/functions.
I'll have a look through the code at some point and see if there are any things that can be made a bit nicer and just make a separate PR if it looks like it is worth changing anything.

**Comment by teonbrooks** [comment]
> Edit: Ok, there were a few little things to fix, nothing major, but I made it a little bit cleaner I think... Should I make a separate PR or something else? (it's only a few small changes...)(cf. [monkeyman192@6fdfe2b](https://github.com/monkeyman192/mne-bids/commit/6fdfe2b8c30e130bf6d23f7442e169be2e715728))

@monkeyman192 yes, a separate pr would be great

**Pull request merged**